daemon/response.go

```diff
@@ -100,12 +94,8 @@ func (r *resp) ServeHTTP(w http.ResponseWriter, _ *http.Request) {
 
 	hdr := w.Header()
 	if r.Status == http.StatusAccepted || r.Status == http.StatusCreated {
-		if m, ok := r.Result.(map[string]interface{}); ok {
-			if location, ok := m["resource"]; ok {
-				if location, ok := location.(string); ok && location != "" {
-					hdr.Set("Location", location)
-				}
-			}
+		if r.Meta.Change != "" {
+			hdr.Set("Location", changeURL(r.Meta.Change))
```

Comment:
That sentence asks us to do exactly what was already being done. In the response we include the change URL which is a pointer to the status. This is not what the `Location` header usually means. In fact, right before the quoted sentence we can read:

> The request has been fulfilled and resulted in a new resource being created. The newly created resource can be referenced by the URI(s) returned in the entity of the response, with the most specific URI for the resource given by a Location header field. 

Which is exactly what I was pointing out above. The `Location`, if present at all, is supposed to be for the _created resource_, not to a place giving further information about the request.